### PR TITLE
Merge 2/genericpath and 2and3/genericpath

### PR DESCRIPTION
**Pull request by ethanhs**
Hopefully this is what you had in mind @gvanrossum?

### CODE REVIEW

File: stdlib/2and3/genericpath.pyi [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
-def exists(path: str) -> bool: ...
-def isfile(path: str) -> bool: ...
-def isdir(s: str) -> bool: ...
-def getsize(filename: str) -> int: ...
-def getmtime(filename: str) -> float: ...
-def getatime(filename: str) -> float: ...
-def getctime(filename: str) -> float: ...
-def commonprefix(m: List[str]) -> str: ...
+if sys.version_info < (3, 0):
+    class _unicode: ...
```
Comment (gvanrossum) [inline stdlib/2and3/genericpath.pyi]: I'd drop this, the comment in the Py2 source says
```
    # If Python is built without Unicode support, the unicode type
    # will not exist. Fake one.
```

File: stdlib/2and3/genericpath.pyi [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
+
+def exists(path: Text) -> bool: ...
+def isfile(path: Text) -> bool: ...
+def isdir(s: Text) -> bool: ...
+def getsize(filename: Text) -> int: ...
+def getmtime(filename: Text) -> float: ...
+def getatime(filename: Text) -> float: ...
+def getctime(filename: Text) -> float: ...
+def commonprefix(m: List[AnyStr]) -> AnyStr: ...
```
Comment (gvanrossum) [inline stdlib/2and3/genericpath.pyi]: commonprefix() should not be generic in Python 3, it only works for strings. (Witness the `return ''`) in the source code.

File: stdlib/2and3/genericpath.pyi (hunk 1 above)
Comment (JelleZijlstra) [reply stdlib/2and3/genericpath.pyi]: Is it even possible to build 2.7 without unicode support? I'm not sure we need to support that.

Comment (gvanrossum) [reply stdlib/2and3/genericpath.pyi]: I guess it once was, but mypy needn't care.

File: stdlib/2and3/genericpath.pyi [hunk 3]
```diff
@@ -1,8 +1,10 @@
 from typing import List, AnyStr, Text
 import sys
 
-if sys.version_info < (3, 0):
-    class _unicode: ...
+if sys.version_info > (3, 0):
+    def commonprefix(m: List[str]) -> AnyStr: ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/genericpath.pyi]: Return type should also be `str`.

Comment (JelleZijlstra) [reply stdlib/2and3/genericpath.pyi]: Also, I think the argument type can be broadened to `Sequence`.

**Pull request merged**